### PR TITLE
Simulcast info

### PR DESCRIPTION
**Pull request by kamil-stasiak**

### CODE REVIEW

File: examples/simple-app/src/main.ts [hunk 1]
```diff
@@ -125,30 +126,97 @@ client.on("peerJoined", (peer) => {
...
+      toastInfo("You cannot set 'targetTrackEncoding' on a track that doesn't have an active simulcast.");
+    }
+  });
+};
+
+const ENCODINGS: TrackEncoding[] = ["l", "m", "h"];
+
 client.on("trackReady", (ctx) => {
   console.log("On track ready");
+  console.log({ name: "trackReady", ctx });
```
Comment (szebniok) [inline examples/simple-app/src/main.ts:160]: Maybe we could join them into a single console.log call

File: examples/simple-app/src/main.ts [hunk 2]
```diff
@@ -125,30 +126,97 @@ client.on("peerJoined", (peer) => {
...
+  if (!ctx.trackId) return;
+
   const peerId = ctx.endpoint.id;
   const peerComponent = document.querySelector(`div[data-peer-id="${peerId}"`)!;
-  const videoPlayer: HTMLVideoElement = peerComponent.querySelector(".remote-peer-template-video")!;
+
+  const videoPlayerTemplate = document.querySelector("#remote-peer-template-video")!;
+
+  // @ts-ignore
+  const videoWrapper = videoPlayerTemplate.content.cloneNode(true);
```
Comment (szebniok) [inline examples/simple-app/src/main.ts:169]:
```suggestion
  const videoPlayerTemplate = document.querySelector<HTMLTemplateElement>("#remote-peer-template-video")!;

  const videoWrapper = <DocumentFragment>videoPlayerTemplate.content.cloneNode(true);
```

All subsequent `querySelector` calls will require non-null assertions. I think that it's still worth it, as the `videoWrapper` is effectively typed as `any` right now.

File: examples/simple-app/src/main.ts [hunk 3]
```diff
@@ -125,30 +126,97 @@ client.on("peerJoined", (peer) => {
...
+  tracksContainer.dataset.trackId = ctx.trackId;
+
+  const videoPlayer: HTMLVideoElement = videoWrapper.querySelector<HTMLVideoElement>(`video`);
+
+  const container = peerComponent.querySelector(".remote-videos");
+
+  if (!container) throw new Error("Remote videos container not found!");
+
+  const simulcastContainer: HTMLDivElement = videoWrapper.querySelector<HTMLDivElement>(`.simulcast-enabled`);
+  simulcastContainer.innerHTML = (ctx?.simulcastConfig?.enabled || false).toString();
```
Comment (szebniok) [inline examples/simple-app/src/main.ts:181]:
```suggestion
  simulcastContainer.innerHTML = (ctx?.simulcastConfig?.enabled ?? false).toString();
```
I think that it is a much cleaner in this case, as we are dealing with a boolean value.

File: examples/simple-app/src/main.ts [hunk 4]
```diff
@@ -125,30 +126,97 @@ client.on("peerJoined", (peer) => {
...
+    activeEncodingElement.innerHTML = ctx.encoding ?? "";
+  });
   ctx.on("voiceActivityChanged", () => {});
 });
 
-client.on("trackRemoved", (_ctx) => {});
+client.on("trackRemoved", (ctx) => {
+  const tracksContainer: HTMLElement | null = document.querySelector(`div[data-track-id="${ctx.trackId}"`);
+  if (!tracksContainer) return;
+  tracksContainer.remove();
```
Comment (szebniok) [inline examples/simple-app/src/main.ts:213]:
```suggestion
  tracksContainer?.remove();
```

File: examples/simple-app/src/main.ts [hunk 5]
```diff
@@ -125,30 +126,97 @@ client.on("peerJoined", (peer) => {
...
+
 client.on("peerUpdated", (_peer) => {});
+
 client.on("peerLeft", (peer) => {
   const peerComponent = document.querySelector(`div[data-peer-id="${peer.id}"`)!;
   peerComponent.remove();
   toastInfo(`Peer left`);
 });
+
+const setUpSimulcastCheckbox = (element: HTMLDivElement, trackId: string, encoding: "l" | "m" | "h") => {
```
Comment (szebniok) [inline examples/simple-app/src/main.ts:138]:
```suggestion
const setupSimulcastCheckbox = (element: HTMLDivElement, trackId: string, encoding: "l" | "m" | "h") => {
```
The `setup` word is already used in this project, so i think it should be preffered here

**Review by szebniok — CHANGES_REQUESTED** [review]

**Review by incubo4u — APPROVED** [review]

**Pull request merged**